src/core/socialprovider.js

```diff
@@ -122,6 +122,14 @@ XMPPSocialProvider.prototype.initializeState = function() {
  * @param {Function} continuation Callback upon connection
  */
 XMPPSocialProvider.prototype.connect = function(continuation) {
+  if (this.client) {
```

Comment:
This can definitely happen if the user clicks one of the login buttons more than once (how I ended up reproducing the mentioned bug). I think it can also happen with reconnect (how I am guessing Izzie reproduced the bug). 

In the first case, it's because the social provider's login method is called more than once. We could stop connect() from being called twice by keeping track of pending or successful login attempts in the UI, core.ts and/or social.ts; however, that would prevent chrome_oauth.ts from being able to open a new auth tab each time the button is clicked (since the tab is launched after the social provider's login method is called).

In the reconnect case, it's possible that the old client is not cleaned up before the new connect attempt starts.

I have a hunch that changing the last few lines of google-auth/facebook-auth to:
```
if (!this.client) {
  this.initializeState();
  this.connect(continuation);
}
```
or changing the beginning of XMPPSocialProvider's onCredentials to:
```
if (msg.cmd && msg.cmd === 'auth') {
  this.credentials = msg.message;
  this.connect(continuation);
}
```
could prevent connect being called multiple times. I felt a little better editing the connect method, but I can explore these changes as well.